### ISSUE TITLE
Weird AppDomain.CurrentDomain.BaseDirectory path in "InProcess" mode

### ISSUE THREAD

**Issue by Tornhoof** [aspnet/IISIntegration#725]
If you run your application "inprocess" in 2.1-preview1-final, ``AppDomain.CurrentDomain.BaseDirectory`` return ``\\?\C:\InProcess`` which apparently breaks file loading in e.g. xmldocument. In my Code there is a File.Exist before that, which works, but then it fails in the doc.Load.

In outofprocess mode it still returns ``c:\InProcess``.

Repro:
run the following snippet in an inprocess web application and copy a test.xml into the folder
```csharp
var fileName = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "test.xml");
var doc = new XmlDocument();
doc.Load(fileName);
```
Exception is:
```ini
System.UriFormatException
  HResult=0x80131537
  Message=Invalid URI: The hostname could not be parsed.
  Source=System.Private.Uri
  StackTrace:
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString, UriKind uriKind)
   at System.Xml.XmlResolver.ResolveUri(Uri baseUri, String relativeUri)
   at System.Xml.XmlUrlResolver.ResolveUri(Uri baseUri, String relativeUri)
   at System.Xml.XmlTextReaderImpl..ctor(String url, XmlNameTable nt)
   at System.Xml.XmlDocument.Load(String filename)
```

Note: This might as well be some bug in corefx, but since it first showed up in the inprocess mode, I'll post it here first. I don't know this path format and apparently not every part of .net core can handle this.

**Comment by Tratcher** [comment]
Sure, but is there a way to suppress the `\\?\` prefix?

**Comment by jkotalik** [comment]
Oh, we can remove the \\?\ portion.

**Comment by Tornhoof** [comment]
Yeah, please remove the prefix, that's all I need I think, I did that manually and at least my app started and loaded the files.
I still have some issues I need to figure out, but I'm one step further now.

**Comment by jkotalik** [comment]
@pan-wang

**Comment by pan-wang** [comment]
the prefix is for  extended-length path and UNC. https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247%28v=vs.85%29.aspx.

**Comment by muratg** [comment]
This is tracking the core-setup bug that @davidfowl linked above.

**Comment by Tornhoof** [comment]
This now breaks fully on a more recent preview2 version, I tried to see if #728 is fixed:
```csharp
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null.
  Source=System.Private.CoreLib
  StackTrace:
   at System.IO.Path.Combine(String path1, String path2)
   at Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute.GetRelatedAssemblies(Assembly assembly, Boolean throwOnError, Func`2 fileExists, Func`2 loadFile) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\ApplicationParts\RelatedAssemblyAttribute.cs:line 95
   at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationAssembliesProvider.GetAssemblyItem(Assembly assembly) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\ApplicationParts\ApplicationAssembliesProvider.cs:line 129
   at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationAssembliesProvider.ResolveFromDependencyContext(DependencyContext dependencyContext) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\ApplicationParts\ApplicationAssembliesProvider.cs:line 87
   at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationAssembliesProvider.<ResolveAssemblies>d__3.MoveNext() in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\ApplicationParts\ApplicationAssembliesProvider.cs:line 56
   at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartManager.PopulateDefaultParts(String entryAssemblyName) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\ApplicationParts\ApplicationPartManager.cs:line 59
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.GetApplicationPartManager(IServiceCollection services) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\DependencyInjection\MvcCoreServiceCollectionExtensions.cs:line 93
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.AddMvcCore(IServiceCollection services) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\DependencyInjection\MvcCoreServiceCollectionExtensions.cs:line 51
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.AddMvcCore(IServiceCollection services, Action`1 setupAction) in D:\b\w\33bdfc1cae7b2a38\modules\Mvc\src\Microsoft.AspNetCore.Mvc.Core\DependencyInjection\MvcCoreServiceCollectionExtensions.cs:line 129
```
Repro you can try yourself:
```csharp
var path = "file:////?/C:/InProcess/MyApp.WebAPI.dll";
if (Uri.TryCreate(path, UriKind.Absolute, out var result) && result.IsFile)
{
    var localPath = result.LocalPath;
    Console.WriteLine(localPath);
    var directoryName = Path.GetDirectoryName(localPath);                
    Console.WriteLine(directoryName);
    if(directoryName == null)
    {
        Console.WriteLine("It's null.");
    }
}
```
Just wondering do you actually have Inprocess tests with MVC applications?
Because as far as I see, the code in the exception is part of ``services.AddMvcCore();`` which is called for every MVC application.
From my side it looks like there are quite a few tests for the inprocess scenario missing.

**Comment by davidfowl** [comment]
If I had to guess I'd say it's to do with the \\?\ path.

**Comment by jkotalik** [comment]
Yeah it is due to the \\\\?\\ path.
@davidfowl we decided on waiting for corefx to fix this, correct?

**Comment by pakrym** [comment]
Lets workaround this in our code. It would unblock enabling MusicStore tests and when we get a fix we would have what to validate it against.

**Comment by Tratcher** [comment]
@jkotalik is this prefix on the path you pass in when initializing the app?

**Comment by jkotalik** [comment]
Yes. I was going to make a PR for this, however it seemed like we wanted this to be fixed in the CoreClr rather than by us. I'd rather fix this though.

**Comment by Tornhoof** [comment]
I did not open any bug on corefx as my repro shows the same problems on netfx, so I thought it's probably simply unsupported.

**Comment by pakrym** [comment]
Workaround merged on our side, tracking https://github.com/dotnet/core-setup/issues/3896

**Comment by kamilzzz** [comment]
So is it somehow possible to run aspnetcore 2.1.0-rc1 in InProcess mode or we have to wait for fix?

Have been unsuccessfully trying to run InProcess for about 3 hours...

**Comment by Tratcher** [comment]
InProcess support was cut from 2.1.0-rc1. We'll revisit in for 2.2.
